src/index/server/route.ts

```diff
@@ -60,10 +48,19 @@ export function create(server: Server, routeConfiguration: { [s: string]: Type }
 			schema
 		};
 
-		return Promise.resolve(message)
-			.then(server.getPublisher().publish.bind(server.getPublisher()))
-			.then(() => responseWriter(server)(undefined, request, response))
-			.catch((error) => responseWriter(server)(error, request, response));
+		try {
+
+			if (synchronous) {
+				server.getPublisher().validate(schema, request.body);
```

Comment:
Should we separate out the validation from the publisher? Seems a bit weird for the publisher to validate?

Reply:
Splitting this out into a `MessageValidator` class.